jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.cpp

```diff
@@ -6,10 +6,12 @@
 #include <cmath>
 #include <iostream>
 
-// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine
-ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_)
-    : roverStateMachine( stateMachine_ ),
-      mJustDetectedObstacle( false ) {}
+// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine, mRoverConfig, and mPhoebe
+ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_, Rover* rover, const rapidjson::Document& roverConfig)
+    : roverStateMachine( stateMachine_ )
+    , mPhoebe( rover )
+    , mRoverConfig( roverConfig )
+    ,  mJustDetectedObstacle( false ) {}
```

Comment:
When I compile I get a warning here. Switching the order of the initializer list so it matches the order of initialization in the hpp file (switch mPhoebe and mRoverConfig) should make the warning go away. I know it doesn't make a difference but its probably just a best practice to not have warnings